source/extensions/filters/network/thrift_proxy/router/upstream_request.cc

```diff
@@ -66,7 +67,9 @@ void UpstreamRequest::onPoolFailure(ConnectionPool::PoolFailureReason reason, ab
 
   // Mimic an upstream reset.
   onUpstreamHostSelected(host);
-  onResetStream(reason);
+  if (!onResetStream(reason)) {
+    parent_.continueDecoding();
```

Comment:
why is this needed?

Reply:
The `UpstreamRequest::start` stops the decoding on downstream if no upstream connection is available immediately.
In the `UpstreamRequest::onPoolFailure` if we don't close the downstream then need to call the `continueDecoding` to be able to get the further requests, otherwise the downstream will just hang.